src/citrus/reconciler.cljs

```diff
@@ -14,13 +15,53 @@
     (release-fn id))
   (vreset! scheduled? (schedule-fn f)))
 
+(defn citrus-default-handler
+  "Implements Citrus' default event handling (as of 3.2.3).
+
+  This function can be copied into your project and adapted to your needs.
+
+  `events` is expected to be a list of events (tuples):
+
+     [ctrl event-key event-args]"
+  [reconciler events]
+  (let [controllers (gobj/get reconciler "controllers")
+        co-effects (gobj/get reconciler "co_effects")
+        effect-handlers (gobj/get reconciler "effect_handlers")
+        state-atom (gobj/get reconciler "state")]
+    (reset!
+      state-atom
+      (loop [state @reconciler
```

Comment:
Are there any changes in the implementation? Hard to tell from the diff, because the whole thing was moved into a function.

Reply:
I tried to keep the implementation exactly as it was, the only change that might affect things is that I've changed how the queue stores items from storing a function to keeping references to the relevant data structures:


https://github.com/clj-commons/citrus/blob/6bb0479dd21c15107591e5b8234ab8a3533691ab/src/citrus/reconciler.cljs#L65-L67

vs
```clj
(queue-effects! queue [cname event args])
```